core/src/worker/data-feed-l2.ts

```diff
@@ -0,0 +1,191 @@
+import { ethers } from 'ethers'
+import { Worker, Queue, Job } from 'bullmq'
+import { Logger } from 'pino'
+import type { RedisClientType } from 'redis'
+import { Aggregator__factory } from '@bisonai/orakl-contracts'
+import { getAggregators, getL2AddressGivenL1Address } from './api'
+import { State } from './state'
+import { IDataFeedListenerWorkerL2, QueueType } from '../types'
+import {
+  BULLMQ_CONNECTION,
+  DEPLOYMENT_NAME,
+  HEARTBEAT_QUEUE_NAME,
+  REMOVE_ON_COMPLETE,
+  SUBMIT_HEARTBEAT_QUEUE_NAME,
+  DATA_FEED_FULFILL_GAS_MINIMUM,
+  L2_CHAIN,
+  REPORTER_AGGREGATOR_L2_QUEUE_NAME,
+  WORKER_AGGREGATOR_L2_QUEUE_NAME,
+  DATA_FEED_WORKER_L2_STATE_NAME
+} from '../settings'
+import { buildSubmissionRoundJobId } from '../utils'
+import { buildTransaction } from './data-feed.utils'
+import { watchman } from './watchman'
+import { OraklError, OraklErrorCode } from '../errors'
+import { getOperatorAddressL2 } from '../api'
+import { oracleRoundStateCallL2 } from './data-feed-l2.utils'
+
+const FILE_NAME = import.meta.url
+
+/**
+ * Get all active aggregators, create their initial jobs, and submit
+ * them to the [heartbeat] queue. Launch [aggregator] and [heartbeat]
+ * workers.
+ *
+ * @param {RedisClientType} redis client
+ * @param {Logger} pino logger
+ */
+export async function worker(redisClient: RedisClientType, _logger: Logger) {
+  const logger = _logger.child({ name: 'worker', file: FILE_NAME })
+
+  // Queues
+  const heartbeatQueue = new Queue(HEARTBEAT_QUEUE_NAME, BULLMQ_CONNECTION)
```

Comment:
I noticed this but still keeps it there because at line 45 we declare `new state()` which requires that queue as a parameter. How about thinking about this and remove it in the next PR?